pkg/scheduler/framework/plugins/migration/utils.go

```diff
@@ -41,12 +41,16 @@ func PredicateResultToFrameworkStatus(reasons []predicates.PredicateFailureReaso
 		return nil
 	}
 
-	if r := predicates.UnresolvablePredicateExists(reasons); r != nil {
-		return framework.NewStatus(framework.UnschedulableAndUnresolvable, r.GetReason())
+	var failureReasons []string
+	if failureReasons = predicates.UnresolvablePredicateExists(reasons); len(failureReasons) != 0 {
+		return framework.NewStatus(framework.UnschedulableAndUnresolvable, failureReasons...)
```

Comment:
if ```UnresolvablePredicateExists``` returns boolean, then we can do this:

```
code := framework.Unschedulable
if predicates.UnresolvablePredicateExists(reasons) {
  code := framework.UnschedulableAndUnresolvable
}
```

Reply:
SG. The only side effect is that for some predicate/fitler plugin, the chances are some reasons are `UnschedulableAndUnresolvable` and some are just `Unschedulable`, then here we're returning all reasons instead of only `UnschedulableAndUnresolvable` reasons. But it should be good overall.

Reply:
I mean: https://github.com/kubernetes/kubernetes/pull/86022/files#diff-fbf6c2939ce03e3e449cf69766ae18aaR57